packages/jest-config/src/normalize.ts

```diff
@@ -789,6 +790,11 @@ export default function normalize(
         value = oldOptions[key];
         break;
       }
+      case 'timeout': {
+        value =
+          oldOptions[key] === 0 ? MAX_32_BIT_SIGNED_INTEGER : oldOptions[key];
```

Comment:
can we throw if it's `0`? I bet 99.99% of the time, that's not what the user wanted.

Should throw if it's `<= 0`

Reply:
Maybe even throw if it's more than, say, an hour. I don't see there being any use case for more than an hour, and again, I'd guess it was some calculation gone wrong for the user

Reply:
I would like to use 0 to represent the infinity timeout. Similar to other test frameworks. I want an infinity timeout when I debugging. I agree most of the time 10 min is more than enough, but there is a few edge case when need an hour.

jasmine always need a DEFAULT_TIMEOUT_INTERVAL value this is the reason why I set it. I use the max possible 32 bit signed value because I would like to avoid the next issue about pls increase the value x+1.

I can implement the <0 error throwing.

Reply:
@SimenB What is the final statement about the setTimeout = 0 and max value? I would like to implement them in 1 comment

Reply:
What I'm afraid of is somebody messing up their math or something and having ci use a bajillion years timing out.

I'd rather have -1 mean infinity, 0 seems like a math error to me. I might be alone in that though :)

Reply:
Why not just require a positive sign number or Infinity.
You can check on infinity and return MAX_32_BIT_SIGNED_INTEGER for jasmine.

Meaning 0 would be instant timeout?

This might be weird for users if other test frameworks default on the 0 = no timeout behaviour.

Reply:
Is there intentional precedence or just what I'd consider buggy code `if (timeout) {setSomeTimeout()}` since `0` is falsy? Seems more natural to be explicit using the `Infinity` global. That doesn't really translate to json or CLI config, though...

I don't feel too strongly here if making `0` mean infinity is something mocha, jasmine, tap, ava or someone else already does

Reply:
- mocha, tap use 0 to represent infinity timeout
- jasmine need a positive integer.  max 2^32-1 the possible value. `Infinity` is not option for jasmine
- ava has not got default timeout

I am using mocha near 6 years and I never make mistake when I switch off the timeout.

All modern CI listen to /read from the standard output and if no message - mostly in 10 min - it stop the build process.

If everyone search for the option to switch off the timeout then he/she will read the documentation and if 0 is the no timeout option will use it.

I think the current solution 0 = no timeout is match with the industry standard. Jasmine need an exact value so we need to translate the 0 to a big value. The biggest value 2^32-1.

It's up to you what will the final solution, but pls don't overthink a simple problem :)